### PR TITLE
use overlay2 docker graph driver in k8s

### LINKED ISSUE #555: Switch Docker Graph driver to overlay2 or aufs

### PR DESCRIPTION
**Pull request by tanmaykm**
**What this PR does / why we need it**: Uses overlay2 as the docker graph driver instead of overlay.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #555

**Special notes for your reviewer**:

**Release note**:

```release-note
use overlay2 docker graph driver in k8s clusters (instead of overlay)
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@anhowe @JackQuincy any concerns with this change?

File: parts/kubernetesagentcustomdata.yml [hunk 1]
```diff
@@ -147,6 +147,18 @@ write_files:
   content: !!binary |
     {{WrapAsVariable "provisionScript"}}
 
+- path: "/opt/azure/containers/useoverlay2.sh"
```
Comment (anhowe) [inline parts/kubernetesagentcustomdata.yml]: why do you need this script?  You already set this file above.

**Comment by anhowe** [comment]
We will need some bake time, but if overlay2 is an improvement, then this should not be a problem.  Any known issues with overlay2?

File: parts/kubernetesagentcustomdata.yml (hunk 1 above)
Comment (tanmaykm) [reply parts/kubernetesagentcustomdata.yml]: Oh yes, it's not needed. Sorry for the mixup. Will fix it in a bit.

**Comment by jackfrancis** [comment]
@yuvipanda Could you kindly add any thoughts on your manual usage of overlay2? Which versions of kubernetes have you tested this out successfully on?

**Comment by yuvipanda** [comment]
@jackfrancis I've been running a 20-node, 1000 pod with moderate container and image churn for about a month on kubernetes 1.7.2 with manual overlay2 and it's been working great!

**Comment by jackfrancis** [comment]
@yuvipanda Thanks for the anecode! Do you have any experience with this on `v1.6` clusters?

**Pull request merged**